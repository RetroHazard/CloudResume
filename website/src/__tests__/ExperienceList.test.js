import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import ExperienceList from '../components/experience_list';

// Sample JSON data
const sampleData = {
    Experience: [
        {
            company: "Generic Company",
            job_title: "Generic Job Title",
            location: "Generic Location",
            type: "Generic Type",
            start: "Generic Start",
            end: "Generic End",
            logo: "./images/placeholder.png",
            website: "https://genericwebsite.com",
            details: [
                "Generic detail 1",
                "Generic detail 2",
                "Generic detail 3",
                "Generic detail 4"
            ],
            technologies: [
                {
                    name: "Generic Skill 1",
                    logo: "ph:placeholder-fill",
                    website: "https://genericskill1.com"
                },
                {
                    name: "Generic Skill 2",
                    logo: "ph:placeholder-fill",
                    website: "https://genericskill2.com"
                },
                {
                    name: "Generic Skill 3",
                    logo: "ph:placeholder-fill",
                    website: "https://genericskill3.com"
                },
                {
                    name: "Generic Skill 4",
                    logo: "ph:placeholder-fill",
                    website: "https://genericskill4.com"
                }
            ]
        }
    ]
};

// Mock the DataLoader component
jest.mock('../utils/dataLoader', () => ({
    __esModule: true,
    default: ({ children }) => children(sampleData),
}));

describe('ExperienceList Component', () => {
    it('renders correctly', () => {
        const { getByText, getByAltText } = render(<ExperienceList />);

        expect(getByText('Generic Job Title')).toBeInTheDocument();
        expect(getByText('Generic Company')).toBeInTheDocument();
        expect(getByText('Generic Type')).toBeInTheDocument();
        expect(getByText('Generic Start - Generic End')).toBeInTheDocument();
        expect(getByText('Generic Location')).toBeInTheDocument();
        expect(getByAltText('Generic Company logo')).toBeInTheDocument();

        sampleData.Experience[0].details.forEach(detail => {
            expect(getByText(detail)).toBeInTheDocument();
        });

        sampleData.Experience[0].technologies.forEach(skill => {
            expect(getByText(skill.name)).toBeInTheDocument();
        });
    });

    it('matches snapshot', () => {
        const { container } = render(<ExperienceList />);
        expect(container).toMatchSnapshot();
    });
});
