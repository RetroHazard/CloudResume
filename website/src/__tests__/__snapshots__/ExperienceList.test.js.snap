// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExperienceList Component matches snapshot 1`] = `
<div>
  <div
    class="skill-list"
  >
    <div
      class="flex flex-col gap-3"
    >
      <div
        class="flex w-full justify-between gap-2"
      >
        <div
          class="flex gap-4"
        >
          <img
            alt="Generic Company logo"
            class="hidden h-[7rem] w-[7rem] rounded-xl sm:block"
            src="./images/placeholder.png"
          />
          <div
            class="flex flex-col"
          >
            <h3
              class="h3 mb-0 font-extrabold text-content-title max-sm:text-sm sm:text-xs md:text-base"
            >
              Generic Job Title
            </h3>
            <p
              class="mb-0.5 font-semibold leading-snug text-content-subtitle max-sm:text-sm sm:text-xs md:text-base"
            >
              Generic Company
            </p>
            <p
              class="mb-0 font-medium leading-snug text-content-accent max-sm:text-xs sm:text-xs md:text-sm"
            >
              Generic Type
            </p>
            <p
              class="mb-0 font-medium leading-snug text-content-date max-sm:text-xs sm:text-xs md:text-sm"
            >
              Generic Start
               - 
              Generic End
            </p>
            <p
              class="mb-0 font-light leading-snug text-content-accent max-sm:text-xs sm:text-xs md:text-xs"
            >
              Generic Location
            </p>
          </div>
        </div>
        <div
          class="flex flex-wrap gap-3 max-sm:flex-col sm:flex-col"
        >
          <a
            aria-label="Website"
            class="social-link"
            href="https://genericwebsite.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            <iconify-icon
              height="1.25em"
              icon="fa6-solid:globe"
              width="1.25em"
            />
          </a>
        </div>
      </div>
      <div
        class="mb-3 font-normal leading-relaxed max-sm:text-xs sm:text-xs sm:leading-relaxed md:text-sm"
      >
        <ul
          class="list-disc pl-10 pr-5"
        >
          <li>
            Generic detail 1
          </li>
          <li>
            Generic detail 2
          </li>
          <li>
            Generic detail 3
          </li>
          <li>
            Generic detail 4
          </li>
        </ul>
      </div>
      <div
        class="flex flex-col gap-1.5"
      >
        <span
          class="font-medium leading-snug text-content-accent max-sm:text-xs sm:text-xs md:text-sm"
        >
          Technologies:
        </span>
        <div
          class="flex flex-wrap max-sm:gap-1 sm:gap-2 md:gap-3"
        >
          <a
            class="skill-block"
            href="https://genericskill1.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            <iconify-icon
              class="icon-box"
              classname="icon-box"
              height="1.1em"
              icon="ph:placeholder-fill"
              width="1.1em"
            />
            Generic Skill 1
          </a>
          <a
            class="skill-block"
            href="https://genericskill2.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            <iconify-icon
              class="icon-box"
              classname="icon-box"
              height="1.1em"
              icon="ph:placeholder-fill"
              width="1.1em"
            />
            Generic Skill 2
          </a>
          <a
            class="skill-block"
            href="https://genericskill3.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            <iconify-icon
              class="icon-box"
              classname="icon-box"
              height="1.1em"
              icon="ph:placeholder-fill"
              width="1.1em"
            />
            Generic Skill 3
          </a>
          <a
            class="skill-block"
            href="https://genericskill4.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            <iconify-icon
              class="icon-box"
              classname="icon-box"
              height="1.1em"
              icon="ph:placeholder-fill"
              width="1.1em"
            />
            Generic Skill 4
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
